ext/mysqli/tests/bug76386.phpt

```diff
@@ -0,0 +1,82 @@
+--TEST--
+Prepared Statement formatter truncates fractional seconds from date/time column (bug #76386)
+--SKIPIF--
+<?php
+require_once('skipif.inc');
+require_once('skipifemb.inc');
+require_once('skipifconnectfailure.inc');
+?>
+--FILE--
+<?php
+require_once('connect.inc');
+// test part 1 - TIMESTAMP(n)
+$link = new mysqli($host, $user, $passwd, $db, $port, $socket);
+$link->query('DROP TABLE IF EXISTS ts_test;');
+$link->query(
+    'CREATE TABLE ts_test (id bigint unsigned auto_increment primary key, ' .
+    'ts timestamp default current_timestamp(), ts2 timestamp(2) default ' .
+    'current_timestamp(2), ts4 timestamp(4) default current_timestamp(4), ' .
+    'ts6 timestamp(6) default current_timestamp(6)) character set utf8 collate ' .
+    'utf8_general_ci;'
+);
+$link->query(
+    'INSERT INTO ts_test (ts, ts2, ts4, ts6) VALUES ("2018-01-01 11:22:33", ' .
+    '"2018-02-02 11:22:33.77", "2018-03-03 11:22:33.8888", ' .
+    '"2018-04-04 11:22:33.999999");'
+);
+$stmt = $link->prepare('SELECT * FROM ts_test;'); // must be statement
+if ($stmt) {
+    $stmt->execute();
+    $tsid = $ts = $ts2 = $ts4 = $ts6 = null;
+    $stmt->bind_result($tsid, $ts, $ts2, $ts4, $ts6);
+    $stmt->fetch();
+    var_dump($ts, $ts2, $ts4, $ts6);
+    $stmt->free_result();
+} else {
+    echo('[FIRST][FAIL] mysqli::prepare returned false: ' . $link->error . PHP_EOL);
+}
+$link->close();
+
+// test part 2 - TIME(n)
+$link = new mysqli($host, $user, $passwd, $db, $port, $socket);
+$link->query('DROP TABLE IF EXISTS t_test;');
+$link->query(
+    'CREATE TABLE t_test (id bigint unsigned auto_increment primary key, ' .
+    't time default current_time(), t2 time(2) default current_time(2), t4 ' .
```

Comment:
I guess `default current_time` is what it doesn't like. It only initalizes timestamp/datetime automatically.

Reply:
Ah, surely enough. I am a total idiot and thought that was IMPOSSIBLE.
It is just that I am using MariaDB, that DOES support this, so I assumed MySQL does as well.
It is the second time that I run into exactly this kind of fail recently.
Anyway that's why I asked about the "official" dockerfile earlier.